src/routes/maps-of-the-week/[page]/+page.server.ts

```diff
@@ -0,0 +1,86 @@
+import { getSortedMapsOfTheWeekNetlifyData } from "$lib/getMapsOfTheWeekNetlifyData";
+import type { MapOfTheWeek } from "../../../types";
+
+const pageSize = 15;
+
+export type MapsOfTheWeekPagePaginatedSSRData = {
+    mapsOfTheWeek: MapOfTheWeek[];
+    pageSize: number;
+    pageCount: number;
+    currentPage: number;
+}
+
+type LoadFunctionParameter = {
+    params: {
+        page: string;
+    },
+    fetch: typeof fetch,
+}
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export async function load({ fetch, params }: LoadFunctionParameter): Promise<MapsOfTheWeekPagePaginatedSSRData> {
+    // Starts at 1
+    const pageNumber = parseInt(params.page, 10);
+    if (isNaN(pageNumber) || pageNumber < 1 || !isFinite(pageNumber)) {
+        throw new Error("Invalid page number");
+    }
+
+    const startIndex = (pageNumber - 1) * pageSize;
+    const endIndex = pageNumber * pageSize;
+
+    const allMapsOfTheWeekNetlifyData = await getSortedMapsOfTheWeekNetlifyData();
+
+    const pageCount = Math.ceil(allMapsOfTheWeekNetlifyData.length / pageSize);
+
+    const paginatedMapsOfTheWeek = allMapsOfTheWeekNetlifyData.slice(startIndex, endIndex);
+
+    const paginatedFullMapsOfTheWeek = [];
+    // Not Promise.all'ing since that will just get you rate limited from beatsaver
```

Comment:
fwiw The maps and users beatsaver endpoints can be called with up to 50 ids at a time. If you can request all of the data and cache it in a dictionary before this loop you can reduce the number of calls (and time taken overall) greatly.